src/components/demos/write/Demo.js

```diff
@@ -1,7 +1,144 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import styled from 'styled-components'
 
 import Header from '../../shared/Header'
+import { Context, initState } from '../../../utils/context'
+import Input from '../../shared/Input'
+import Box from '../../shared/Box'
+import { Container } from '../../shared/Common'
+
+const Wrapper = styled.div`
+  grid-area: body;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+  ${(props) => props}
+`
+const Button = styled.button((props) => props)
+const LoginAttemptText = styled.p((props) => props)
 
 export default function WriteDemo(props) {
-  return <Header {...props} showClose={true} />
+  const state = React.useContext(Context)
+  const writeDemo = { ...state.writeDemo, ...props }
+  const { update } = state
+  const { theme } = writeDemo
+
+  const [demoOutput, SetDemoOutput] = useState('')
+
+  const usernameUpperBound = 24
+  const passwordUpperBound = 16
+  const [usernameInput, SetUsernameInput] = useState('')
+  const [passwordInput, SetPasswordInput] = useState('')
+  const [someUsernameTyped, SetSomeUsernameTyped] = useState(false)
+  const [somePasswordTyped, SetSomePasswordTyped] = useState(false)
+
+  const noUsernameEntered = usernameInput.length === 0 && someUsernameTyped
+  const usernameAtMaxLength = usernameInput.length === usernameUpperBound
+  const noPasswordEntered = passwordInput.length === 0 && somePasswordTyped
+  const passwordAtMaxLength = passwordInput.length === passwordUpperBound
+
+  const UsernameErrorWarning = () => (
+    <>
```

Comment:
do we need <React.Fragment>? Could it be just an array?

Reply:
I figured both warnings can never happen at the same time so may as well use a fragment rather than list